src/main/java/org/terasology/climateConditions/BodyTemperatureSystem.java

```diff
@@ -0,0 +1,124 @@
+import org.terasology.logic.characters.AliveCharacterComponent;
+import org.terasology.logic.chat.ChatMessageEvent;
+import org.terasology.logic.delay.DelayManager;
+import org.terasology.logic.delay.PeriodicActionTriggeredEvent;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.registry.In;
+import org.terasology.world.WorldComponent;
+
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class BodyTemperatureSystem extends BaseComponentSystem {
+    public static final String BODY_TEMPERATURE_UPDATE_ACTION_ID = "Body Temperature Update";
+
+    private static final Logger logger = LoggerFactory.getLogger(BodyTemperatureSystem.class);
+
+    @In
+    private EntityManager entityManager;
+    @In
+    ClimateConditionsSystem climateConditionsSystem;
+    @In
+    DelayManager delayManager;
+
+    private static final int CHECK_INTERVAL = 1000;
+
+    public void postBegin() {
+        boolean processedOnce = false;
+        for (EntityRef entity : entityManager.getEntitiesWith(WorldComponent.class)) {
+            if (!processedOnce) {
+                delayManager.addPeriodicAction(entity, BODY_TEMPERATURE_UPDATE_ACTION_ID, 0, CHECK_INTERVAL);
+                processedOnce = true;
+            } else {
+                logger.warn("More than one entity with WorldComponent found");
+            }
+        }
+        // The world is the entity here which shall have a Body Temperature Update Periodic Action.
+    }
+
+    @ReceiveEvent
+    public void onTemperatureUpdate(PeriodicActionTriggeredEvent event, EntityRef world) {
+        if (event.getActionId().equals(BODY_TEMPERATURE_UPDATE_ACTION_ID)) {
+            for (EntityRef entity : entityManager.getEntitiesWith(AliveCharacterComponent.class,
+                    BodyTemperatureComponent.class, LocationComponent.class)) {
+                LocationComponent location = entity.getComponent(LocationComponent.class);
+                BodyTemperatureComponent bodyTemperature = entity.getComponent(BodyTemperatureComponent.class);
+                float envTemperature = climateConditionsSystem.getTemperature(location.getLocalPosition().getX(),
+                        location.getLocalPosition().getY(), location.getLocalPosition().getZ());
+                float envHumidity = climateConditionsSystem.getHumidity(location.getLocalPosition().getX(),
+                        location.getLocalPosition().getY(), location.getLocalPosition().getZ());
+                float deltaTemp =
+                        ((((envTemperature - (envHumidity / 10)) - bodyTemperature.current) / 100000) * CHECK_INTERVAL);
+                /*
+                //Send event for other systems to modify change in body temperature.
+                AffectBodyTemperatureEvent affectBodyTemperatureEvent = new AffectBodyTemperatureEvent(deltaTemp);
+                entity.send(affectBodyTemperatureEvent);
+                deltaTemp = affectBodyTemperatureEvent.getResultValue();
```

Comment:
>  deltaTemp = affectBodyTemperatureEvent.getResultValue();

That's because here for some reason I was getting some weird values as in for a few seconds it would return an expected value of deltaTemp however after then for a few seconds deltaTemp would simply become zero and there'll be no change and then the oscialltion continues between an expected value and zero every few seconds. I haven't pinpointed the problem yet but I'll check that again.

Reply:
That is weird 🧐 Either we're missing something about how the events are handled OR something is modifying/"canceling" the `AffectBodyTemperatureEvent` - which would make me wonder where that should be happening...

Reply:
I don't know... but I checked and the AffectBodyTemperatureEvent itself is not being used anywhere else..

Reply:
fixed. AbstractValueModifiableEvent.getResultValue() at some point removes negative values..

Reply:
I see - `AbstractValueModifiableEvent` has that "documented" as line comment but not as method contract 🙄  will fix that...

Reply:
Created https://github.com/MovingBlocks/Terasology/pull/4063 to discuss different options to go forward. Please join the discussion 🙃